climada/engine/uncertainty/base.py

```diff
@@ -0,0 +1,728 @@
+        def litpop_cat(m, n):
+            exp = Litpop()
+            exp.set_country('CHE', exponent=[m, n])
+            return exp
+        distr_dict = {
+            m: sp.stats.randint(low=0, high=5),
+            n: sp.stats.randint(low=0, high=5)
+            }
+        unc_var_cat = UncVar(litpop_cat, distr_dict)
+
+    Continuous variable function: Impact function for TC
+        import scipy as sp
+        def imp_fun_tc(G, v_half, vmin, k, _id=1):
+            imp_fun = ImpactFunc()
+            imp_fun.haz_type = 'TC'
+            imp_fun.id = _id
+            imp_fun.intensity_unit = 'm/s'
+            imp_fun.intensity = np.linspace(0, 150, num=100)
+            imp_fun.mdd = np.repeat(1, len(imp_fun.intensity))
+            imp_fun.paa = np.array([sigmoid_function(v, G, v_half, vmin, k)
+                                    for v in imp_fun.intensity])
+            imp_fun.check()
+            impf_set = ImpactFuncSet()
+            impf_set.append(imp_fun)
+            return impf_set
+        distr_dict = {"G": sp.stats.uniform(0.8, 1),
+              "v_half": sp.stats.uniform(50, 100),
+              "vmin": sp.stats.norm(loc=15, scale=30),
+              "k": sp.stats.randint(low=1, high=9)
+              }
+        unc_var_cont = UncVar(imp_fun_tc, distr_dict)
+
+    """
+
+    def __init__(self, unc_var, distr_dict):
+        """
+        Initialize UncVar
+
+        Parameters
+        ----------
+        unc_var : function
+            Variable defined as a function of the uncertainty parameters
+        distr_dict : dict
+            Dictionary of the probability density distributions of the
+            uncertainty parameters, with keys the matching the keyword
+            arguments (i.e. uncertainty parameters) of the unc_var function.
+            The distribution must be of type scipy.stats
+            https://docs.scipy.org/doc/scipy/reference/stats.html
+
+        Returns
+        -------
+        None.
+
+        """
+        self.labels = list(distr_dict.keys())
+        self.distr_dict = distr_dict
+        self.unc_var = unc_var
+
+
+    def plot(self):
+        """
+        Plot the distributions of the parameters of the uncertainty variable.
+
+        Returns
+        -------
+        fig, ax: matplotlib.pyplot.figure, matplotlib.pyplot.axes
+            The figure and axis handle of the plot.
+
+        """
+        # zip_longest(flat_axes, cols, fillvalue=None):
+
+        nplots = len(self.distr_dict)
+        nrows, ncols = int(nplots / 3) + 1, min(nplots, 3)
+        fig, axis = plt.subplots(nrows=nrows, ncols=ncols, figsize=(20, 16))
+        if nplots > 1:
+            flat_axes = axis.flatten()
+        else:
+            flat_axes = [axis]
+        for ax, name_distr in zip_longest(flat_axes,
+                                    self.distr_dict.items(), 
+                                    fillvalue=None):
+            if name_distr is None:
+                ax.remove()
+                continue
```

Comment:
When there is nothing to plot, the outer loop should continue without plotting (i.e. not go to line 132...), otherwise one would get an error. 
What this basically here does, is remove the extra canvases at the end. Say you have 4 distributions. Then you will have 2 rows of 3 plots. But the 5th and 6th panels would be empty plots. With the code here, I remove the axis entirely. I could also write the same as an if/else block. Would you prefer that?

Reply:
Ah, now I understand what the `continue` statement does. If line 129 evaluates to `True`, the `continue` makes the code continue with the next "item" in the for-loop instead of progressing to line 132.
Well.. It's neat in a sense, but I think a bit advanced. The if/else syntax is certainly easier to immediately understand for the average person looking at our source code, but we can also leave it this way, since it's not against any convention.

Reply:
Exactly! Is it too convoluted? I found it easier to read that big if...else.. blocks, but that depends on the familiarity with the `continue` statement. Should I change it?

Reply:
hmm. I don't know. I also agree that it's nice not to have huge semi-repetitive if/else blocks.. I think it's a tie with the fact that it is a bit convoluted if one doesnt know the continue syntax.